api/handlers/service_offering_test.go

```diff
@@ -37,6 +37,123 @@ var _ = Describe("ServiceOffering", func() {
 		routerBuilder.LoadRoutes(apiHandler)
 	})
 
+	Describe("GET /v3/service_offering/:guid", func() {
+		BeforeEach(func() {
+			serviceOfferingRepo.GetServiceOfferingReturns(repositories.ServiceOfferingRecord{
+				ServiceOffering: services.ServiceOffering{},
+				CFResource: model.CFResource{
+					GUID: "offering-guid",
+				},
+				ServiceBrokerGUID: "broker-guid",
+			}, nil)
+		})
+
+		JustBeforeEach(func() {
+			req, err := http.NewRequestWithContext(ctx, "GET", "/v3/service_offerings/offering-guid", nil)
+			Expect(err).NotTo(HaveOccurred())
+
+			routerBuilder.Build().ServeHTTP(rr, req)
+		})
+
+		It("returns the service offering", func() {
+			Expect(serviceOfferingRepo.GetServiceOfferingCallCount()).To(Equal(1))
+			_, actualAuthInfo, actualOfferingGUID := serviceOfferingRepo.GetServiceOfferingArgsForCall(0)
+			Expect(actualAuthInfo).To(Equal(authInfo))
+			Expect(actualOfferingGUID).To(Equal("offering-guid"))
+
+			Expect(rr).Should(HaveHTTPStatus(http.StatusOK))
+			Expect(rr).To(HaveHTTPHeaderWithValue("Content-Type", "application/json"))
+			Expect(rr).To(HaveHTTPBody(SatisfyAll(
+				MatchJSONPath("$.guid", "offering-guid"),
+			)))
+		})
+
+		Describe("include broker field", func() {
+			BeforeEach(func() {
+				serviceBrokerRepo.ListServiceBrokersReturns([]repositories.ServiceBrokerRecord{{
+					ServiceBroker: services.ServiceBroker{
+						Name: "broker-name",
+					},
+					CFResource: model.CFResource{
+						GUID: "broker-guid",
+					},
+				}}, nil)
```

Comment:
Not sure if we need this, since we already test the include broker field in the ListServiceOfferings test.

Reply:
Yeah, looks a bit of a duplication. I would recommend having two whens: `When("including broker name"` and `When("including broker guid` that verify that proper broker fields are being returns. Error cases, as you have pointed out, are already covered in other tests.

Reply:
Done

Reply:
After we use the generic include resolver as advised above testing becomes simpler too, for [example](https://github.com/cloudfoundry/korifi/blob/df9925de6b12eeb51cc05d3dd70019250b8e0175/api/handlers/service_plan_test.go#L126-L153)